### ISSUE TITLE
Git Synchronize changes doesn't work

### ISSUE THREAD

**Issue by elyohan14** [microsoft/vscode#137172]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.62.2
- OS Version: Ubuntu 20.04

Steps to Reproduce:

1. I did changes in repository files
2. I click on synchronize files to pull and push commits
3. This is the error output
```
> git pull --tags origin dev
remote: HTTP Basic: Access denied
fatal: Authentication failed for 'https://gitlab.com/juantorbay/agripac-acua.git/'
```
4. This is the other error: 
```git log --oneline --cherry dev...dev@{upstream} --
> git pull --tags origin dev
remote: HTTP Basic: Access denied
fatal: Authentication failed for 'https://gitlab.com/juantorbay/agripac-acua.git/'
```

**Comment by pintariching** [comment]
I get the same error when I try to clone a repository.

- VS Code Version: 1.62.2
- OS Version: Pop!_OS 21.04

Steps to Reproduce:

1. Click Clone Repository
2. Select a private Repository and allow access in the browser pop-up
3. I get this error
```
> git clone https://github.com/pintariching/DelavnicaNET.git /home/tilen/Documents/DelavnicaNET --progress
remote: Repository not found.
fatal: Authentication failed for 'https://github.com/pintariching/DelavnicaNET.git/'
```

I fixed the problem temporarily by following the tutorial on this website: [https://florianwinkelbauer.com/posts/2021-05-27-git-gnome-keyring/](https://florianwinkelbauer.com/posts/2021-05-27-git-gnome-keyring/).

**Comment by dannyb2018** [comment]
I had same thing pop up on Friday 12th November 2021 except it was with Github and I fixed it by getting a token issued by github and adding it to git credentials on my local machine

**Comment by hdocmsu** [comment]
Hi all, I have the same problem, I did try the method mentioned by @pintariching but it did not work. @dannyb2018, could you please describe the steps you took that solved the problem? Thanks.

FYI: it worked fine for me two days ago but I encounter the error yesterday. here is the error I see from my computer (Ubuntu Linux 21.04, VS code 1.62.2):

git push

remote: Repository not found.
fatal: Authentication failed for ...

**Comment by ashirsat** [comment]
Hi,

I had the same issue with vscode when I was trying to clone a private repository to my local machine. 

Ubuntu: 18.04 LTS
VSCode version: 1.62.2

The error also occurs for other git workflows like pull,push as well.

The error for the git workflows are as follows:
git clone 

```shell
remote: Repository not found.
fatal: Authentication failed for:
```

git pull , git push
```shell
remote: Repository not found.
fatal: Authentication failed for
```

**Comment by dannyb2018** [comment]
1) Remove the credentials for git that you currently have

2) Add libsecret:
sudo apt-get install libsecret-1-0 libsecret-1-dev
cd /usr/share/doc/git/contrib/credential/libsecret
sudo make

3) make libsecret the credential helper: 
git config --global credential.helper /usr/share/doc/git/contrib/credential/libsecret/git-credential-libsecret

4) Get a token for logging into Github:
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

4) Do a git push and it will ask for user name - use your normal one - and when it asks for the password give the token

So far so good for me.  No idea why it stopped working first

**Comment by hdocmsu** [comment]
@dannyb2018: Thanks a lot for your prompt response!

a follow-up question, could you please elaborate on the 1st step a little more?

How did you remove the credential? is there an instruction to do it on VS code or a linux command for it? additional info would be appreciated. I think I perform steps 2-5 fine.

Thanks!

**Comment by pintariching** [comment]
I think that he means that by removing credentials, you can use either a GUI tool like `seahorse` or a CLI tool like `secret-tool` (you can get them with `sudo apt-get install seahorse` or `sudo apt-get install secret-tool`)

Some distributions already have these installed and you can find them if you search for something like `keyring`.

**Comment by hdocmsu** [comment]
Thanks, @dannyb2018 and @pintariching for your response. I did try your suggestions and a few other random things from stackoverflow.com that I cannot recall and reproduce what I did including restarting my computer, uninstallation, and reinstallation of GitHub extensions. looks like git push works fine for me now. 

Hope that someone could reproduce/confirm the fix and share it here so that others could be benefited from it.

**Comment by bobonis** [comment]
I'm facing the same issue, starting from today after I've updated to latest VScode version. I'm not using gnome-keyring or seahorse to store git credentials.

 - VS Code Version: 1.62.2
 - OS Version: Ubuntu 20.04

Pulling from private gitlab server using gnome-terminal concludes successfully. Pulling from VSCode terminal results in error:
```
remote: HTTP Basic: Access denied
```
Seahorse does not have entries for git credentials.

**Comment by gjsjohnmurray** [comment]
/duplicate of #137183 which is scheduled to be fixed in upcoming recovery release (1.62.3)

**Comment by elyohan14** [comment]
I did change from HTTPS to SSH url and it works for me!
